extensions/amp-fit-text/0.1/amp-fit-text.js

```diff
@@ -236,6 +219,47 @@ export function updateOverflow_(content, measurer, maxHeight, fontSize) {
   });
 }
 
+/**
+ *
+ * @param {!Document} document
+ * @param {!Element} element
+ */
+export function buildDom(document, element) {
```

Comment:
I think it would be useful to have a standard comment block for `buildDom` functions explaining their limitations and use. As an aside, it may even be reasonable to create a dedicated `build-dom.js` in each extension directory since a) it allows importing the function without the side-effects of registering the extension at the bottom of the file and b) it would allow an OWNERS rule requiring/alerting you to any changes, since it would be easy to introduce compiler-incompatible code.

Reply:
I agree completely, and the first incarnation of buildDom was actually in its own file for exactly this purpose. Justin pushed back initially because co-locating is a bit nicer from readability standpoint.

I'd prefer to continue to hold off on moving it to its own file unless we end up really needing to.

Reply:
Also the ability to selectively enforce more strict lint rules, forbidden terms, etc. @jridgewell this seems to be in line with the style of changes we've already been making to split apart component code for compiling different NPM binaries. As written, would it even be possible to bundle `buildDom` into a compiler binary without pulling in the entire element definition as a result of the side-effect-ful registration at the bottom of this file?